### PR TITLE
Improve `saved_model.load.get_config`

### PR DESCRIPTION
**Pull request by MiWeiss**
This PR makes two minor, nonfunctional improvements to the `saved_model.load.get_config` function:

1. **it’s easier to ask for forgiveness than permission:**  
Idiomatic python: It's better to catch the exception of the exceptional case, rather than checking if the attribute exists beforehand. [more on this](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/)
2.  **Error description:** 
The NotImplementedError now describes two likely root causes of the problem.

Also opened this in the tensorflow repo, before noticing this probably belongs here:
https://github.com/tensorflow/tensorflow/pull/53457

### CODE REVIEW

File: keras/saving/saved_model/load.py [hunk 1]
```diff
@@ -1057,10 +1057,15 @@ def keras_api(self):
...
 
   def get_config(self):
-    if hasattr(self, '_config'):
+    try:
       return self._config
-    else:
-      raise NotImplementedError
+    except AttributeError as e:
+      raise NotImplementedError(
+        "`_config` not found on layer. "
```
Comment (fchollet) [inline keras/saving/saved_model/load.py:1064]: "`_config` not found on layer." is not a message that end users can understand. We need to provide better context.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Please also add a unit test that uses `assertRaises` to check that the error message is displayed as expected.

**Comment by gbaned** [comment]
@MiWeiss Can you please check @fchollet's comments and keep us posted ? Thanks!

File: keras/saving/saved_model/load.py (hunk 1 above)
Comment (MiWeiss) [reply keras/saving/saved_model/load.py:1064]: I agree. That's why I gave the context on the next couple of lines (there would not be a line break once raised). Not sure if you noticed. But I guess you're right in either case and I should rephrase s.t. the higher level info comes first.

In either case, as this PR is probably going to be closed, never mind :-)

**Comment by MiWeiss** [comment]
As indicated by @fchollet, this is missing an adaption the test which checks for the raised exception to also check for the new message, which should be reasonable quick.

However, it appears that not only there is no such test, but moreover the file `keras.saving.saved_model` has no test class yet. At the moment, I do not have sufficient spare time to familiarize myself enough with saved_model internals to set up such a class, Thus, @fchollet @gbaned feel free to close or keep open as you prefer.

**Pull request closed without merging**